src/OrleansRuntime/Streams/QueueBalancer/DeploymentBasedQueueBalancer.cs

```diff
@@ -179,17 +137,11 @@ public IEnumerable<QueueId> GetMyQueues()
             bool useIdealDistribution = isFixed || isStarting;
             Dictionary<string, List<QueueId>> distribution = useIdealDistribution
                 ? balancer.IdealDistribution
-                : balancer.GetDistribution(GetActiveSilos(siloStatusOracle, immatureSilos));
+                : balancer.GetDistribution(GetActiveSilos(siloStatusOracle));
 
             List<QueueId> myQueues;
             if (distribution.TryGetValue(siloStatusOracle.SiloName, out myQueues))
             {
-                if (!useIdealDistribution)
-                {
-                    HashSet<QueueId> queuesOfImmatureSilos = GetQueuesOfImmatureSilos(siloStatusOracle, immatureSilos, balancer.IdealDistribution);
-                    // filter queues that belong to immature silos
-                    myQueues.RemoveAll(queue => queuesOfImmatureSilos.Contains(queue));
-                }
```

Comment:
This part also confuses me. 
In line 182, `GetActiveSilos` excluded immature silos from `activeSilos` list returned by `siloStatusOracle`, which would make current silo take some of immature silos' queues in the `distrubution`.
But in line 191, the code filtered out queues from immature silos in `myQueues`.

To me, it is like doing a + followed by a -, which cancels each other. it seems to give the same outcome if we do:  get `activeSilos` from `siloStatusOracle` ->distribution = balancer.GetDistribution(activeSilos)-> return current silo's queues from the distribution. 

And this seems to be the only part where `immatureSilos` (track immaturity for other silos) contribute to, which go back to my original confusion: why do we need to maintain that `immatureSilos` dic?